Reefer Operational Monitoring Notification API

cmsdroff opened the pull request:
First draft for the RMO_NTF api to provide shallow notifications to be provided to carriers about reefer events.

HenrikHL commented on rmo/notification/v1/rmo_ntf_v1.0.0.yaml:
```diff
@@ -0,0 +1,268 @@
+  description: |
+    API specification issued by DCSA.org.
+
+    Operational Reefer Monitoring Notifications for [DCSA OpenAPI specification for Operational Reefer Monitoring](https://app.swaggerhub.com/apis/dcsaorg/DCSA_RMO/1.0.0-Beta-1) is a lightweight notification based on [CloudEvents](https://cloudevents.io/). The `POST` endPoint of the consumer is called whenever a `Reefer` that is being subscribed to changes state or is updated.
```
Does the Reefer change state? In Booking - the Booking actually changes state from `RECEIVED` to `PENDING UPDATE` - that is why I have used that wording... It might also be correct for Reefer?

cmsdroff replied:
I've modified and will clarify with Alex, your correct to question as when I read back maybe state isn't the right word, in the Reefer Monitoring API the consumer will subscribe to agreed notification types with their provider it maybe an alarm being activated on the reefer, or a geofence being triggered.  I've re-worded and submitting change.

HenrikHL commented on rmo/notification/v1/rmo_ntf_v1.0.0.yaml:
```diff
@@ -0,0 +1,268 @@
+    ### Stats API
+    The Stats API offers crucial statistical information for both API providers and consumers to enhance their services and helps DCSA to understand and scale the ecosystem. We expect you to invoke the Stats API for every request made to the Operational Reefer Monitoring Notification API. Further details can be found [here](https://developer.dcsa.org/#/http/guides/api-guides/stats-api)
+
+    For a changelog please click [here](https://github.com/dcsaorg/DCSA-OpenAPI/tree/master/rmo/notification/v1). Please also [create a GitHub issue](https://github.com/dcsaorg/DCSA-OpenAPI/issues/new) if you have any questions/comments.
```
```suggestion
    For a changelog please click [here](https://github.com/dcsaorg/DCSA-OpenAPI/tree/master/rmo/notification/v1#v100B1). Please also [create a GitHub issue](https://github.com/dcsaorg/DCSA-OpenAPI/issues/new) if you have any questions/comments.
```

HenrikHL commented on rmo/v1/reefer_v1.0.0-Beta-1.yaml:
```diff
@@ -0,0 +1,807 @@
+
+    Calling this endPoint will return a list of events that fulfill the query parameters specified. The list returned is sorted in ascending (`ASC`) order based on the creation date-time of the event (`eventCreatedDateTime`). **This endPoint is mandatory to implement.**
+
+    For a changelog please click [here](https://github.com/dcsaorg/DCSA-OpenAPI/blob/master/reefer-monitoring/v1#v1000B1). Please also [create a GitHub issue](https://github.com/dcsaorg/DCSA-OpenAPI/issues/new) if you have any questions/comments.
```
```suggestion
    For a changelog please click [here](https://github.com/dcsaorg/DCSA-OpenAPI/blob/master/reefer-monitoring/v1#v100B1). Please also [create a GitHub issue](https://github.com/dcsaorg/DCSA-OpenAPI/issues/new) if you have any questions/comments.
```

HenrikHL replied:
my bad...

HenrikHL commented on rmo/notification/v1/rmo_ntf_v1.0.0.yaml:
```diff
@@ -0,0 +1,268 @@
+      summary: Send a new Operational Reefer Notification
+      operationId: operational-reefer-notifications
+      description: |
+        Creates a new [`Operational Reefer Notification`](#/OperationalReeferNotification). This endPoint is called whenever a `Reefer` that a consumer has subscribed to changes state or is updated.
```
Same question regarding the wording `state`...

HenrikHL commented on rmo/notification/v1/rmo_ntf_v1.0.0.yaml:
```diff
@@ -0,0 +1,268 @@
+          required: true
+          schema:
+            type: string
+            example: 2.0.0-Beta-1
```
```suggestion
            example: 1.0.0-Beta-1
```

HenrikHL commented on rmo/notification/v1/rmo_ntf_v1.0.0.yaml:
```diff
@@ -0,0 +1,268 @@
+                      alarmStatus: ACTIVE
+                      alarmNumber: '100'
+                      alarmSeverity: CRITICAL
+              declinedExample:
```
```suggestion
              deactivedExample:
```

HenrikHL commented on rmo/notification/v1/rmo_ntf_v1.0.0.yaml:
```diff
@@ -0,0 +1,268 @@
+                [SemVer](https://semver.org/) used to indicate the version of the consumer (the responding party). This is optional to provide.
+              schema:
+                type: string
+                example: 2.0.0-Beta-1
```
```suggestion
                example: 1.0.0-Beta-1
```

HenrikHL commented on rmo/v1/reefer_v1.0.0-Beta-1.yaml:
```diff
@@ -0,0 +1,807 @@
+
+    Calling this endPoint will return a list of events that fulfill the query parameters specified. The list returned is sorted in ascending (`ASC`) order based on the creation date-time of the event (`eventCreatedDateTime`). **This endPoint is mandatory to implement.**
+
+    For a changelog please click [here](https://github.com/dcsaorg/DCSA-OpenAPI/blob/master/reefer-monitoring/v1#v100B1). Please also [create a GitHub issue](https://github.com/dcsaorg/DCSA-OpenAPI/issues/new) if you have any questions/comments.
```
```suggestion
    For a changelog please click [here](https://github.com/dcsaorg/DCSA-OpenAPI/blob/master/rmo/v1#v100B1). Please also [create a GitHub issue](https://github.com/dcsaorg/DCSA-OpenAPI/issues/new) if you have any questions/comments.
```

cmsdroff commented:
@HenrikHL I have completed work on the RMO and RMO_NTF in stoplight.

- Removed the full type notifications as discussed,
- Updated with POST for the RMO api to provide reefer data to carriers
- Updated the Readme to match for consistency.
- Merged master to RMO_NTF

Im good to prepare to finalise our work on this, if you could give it the eyes over and push to swagger hub the unpublished API specs would be much appreciated.

HenrikHL commented on rmo/notification/v1/rmo_ntf_v1.0.0-Beta-1.yaml:
```diff
@@ -0,0 +1,271 @@
+                id: 23fcnasr894y0
+              description: Provided when alarm is activated deactivated on the reefer.
+              properties:
+                alarmStatus:
+                  type: string
+                  x-stoplight:
+                    id: h63fl95ba6npl
+                  description: |-
+                    Indicates if the alarm is active or inactive.  Possible values are:
+                    * `ACTIVE` meaning the alarm has been activated.
+                    * `INACTIVE` meaning the alarm is no longer active.
```
```suggestion
                alarmStatus:
                  type: string
                  x-stoplight:
                    id: h63fl95ba6npl
                  description: |-
                    Indicates if the alarm is active or inactive.  Possible values are:
                    * `ACTIVE` meaning the alarm has been activated.
                    * `INACTIVE` meaning the alarm is no longer active.
                  example: ACTIVE
```

HenrikHL commented on rmo/notification/v1/rmo_ntf_v1.0.0-Beta-1.yaml:
```diff
@@ -0,0 +1,271 @@
+servers:
+  - url: 'http://localhost:3000'
+paths:
+  /v2/operational-reefer-notifications:
```
```suggestion
  /v1/operational-reefer-notifications:
```

HenrikHL commented on rmo/v1/rmo_v1.0.0-Beta-1.yaml:
```diff
@@ -0,0 +1,835 @@
+    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
+    name: Apache 2.0
+paths:
+  /events:
```
```suggestion
  /v1/events:
```
